daemon/container_operations.go

```diff
@@ -615,6 +615,18 @@ func validateEndpointSettings(nw *libnetwork.Network, nwName string, epConfig *n
 		}
 	}
 
+	if sysctls, ok := epConfig.DriverOpts[netlabel.EndpointSysctls]; ok {
+		for _, sysctl := range strings.Split(sysctls, ",") {
```

Comment:
Naively using comma as a delimiter without any escaping could come to bite us. While there are no documented per-iface sysctls I could find which have a comma-separated list as the value, [there is existing precedent in other sysctls.](https://www.kernel.org/doc/html/latest/networking/ip-sysctl.html#:~:text=ip_local_reserved_ports%20%2D%20list%20of%20comma%20separated%20ranges) While we could come up with a microformat which allows commas in values to be quoted or escaped, that seems like a lot of fuss. Could there be a better way? (Apologies if I'm rehashing some stuff we discussed earlier.)

The problem is with packing multiple sysctls into a single DriverOpts value. What if we instead encoded each sysctl as a _distinct_ DriverOpt so that the keys and values are already structured?
```json
{
  "com.docker.endpoint.sysctl.net.ipv4.conf.arp_announce": "1",
  "com.docker.endpoint.sysctl.net.ipv6.conf.dad_transmits": "4"
}
```

The only downside to this scheme is that it leaves the order that sysctls are applied undefined unless we take steps to define an ordering. As applying per-iface sysctls out-of-order is currently only a theoretical issue, we can probably get away with defining an arbitrary _consistent_ order that is independent of Go's map iteration randomization. (I have a hunch that sorting lexicographically by path, — splitting the sysctl key on periods and sorting by the resulting tuples — would yield a different and more intuitive order than a naive `strings.Sort`, but I have not yet found a counterexample.)

YAGNI, but if user-controlled sysctl order is actually required, we could afford users a way to provide an explicit sort order discriminant, kinda like the classic "prefix your init scripts with a number to control the order they are invoked." For example:
```json
{
  "com.docker.endpoint.sysctl.net.ipv6.conf.forwarding": "1",
  "com.docker.endpoint.sysctl[42].net.ipv6.conf.accept_redirects": "1",
  "com.docker.endpoint.sysctl[99].net.ipv6.conf.accept_ra": "2"
}
```

Reply:
Right, yes - there's some chance that maybe one day there might be a sysctl value format that causes problems.

In the meantime, it's a trade-off. Per-sysctl options would unnecessarily make commands more verbose right now, and sacrifice the ordering. The ordering isn't important at the moment - but I don't think we can predict with any certainty whether future sysctls will introduce comma-formatting or a requirement for ordering.

`docker run --network='name=bad="driver-opt=com.docker.network.endpoint.sysctls=ipv4.conf.log_martians=1,ipv4.conf.forwarding=0"' ...`

`docker run --network=name=worse=driver-opt=com.docker.network.endpoint.sysctl.ipv4.conf.log_martians=1,driver-opt=com.docker.network.endpoint.sysctl.ipv4.conf.forwarding=0 ...`

So, for the foreseeable future, I think it's better as-is. Most likely YAGNI but, if the situation does arise, per-sysctl options can be implemented as an alternative at that point with no issues for backwards compatibility.

Reply:
SGTM 👍